Fluid/Ast/ForStatement.cs

```diff
@@ -147,5 +163,66 @@ public override async Task<Completion> WriteToAsync(TextWriter writer, TextEncod
 
             return Completion.Normal;
         }
+
+        private class ForLoopValue : FluidValue
+        {
+            public int Length { get; set; }
+            public int Index { get; set; }
+            public int Index0 { get; set; }
+            public int RIndex { get; set; }
+            public int RIndex0 { get; set; }
+            public bool First { get; set; }
+            public bool Last { get; set; }
+
+            public int Count => Length;
+
+            public override FluidValues Type => FluidValues.Dictionary;
+
+            public override bool Equals(FluidValue other)
+            {
+                return false;
+            }
+
+            public override bool ToBooleanValue()
+            {
+                return false;
+            }
+
+            public override double ToNumberValue()
+            {
+                return Length;
+            }
+
+            public override object ToObjectValue()
+            {
+                return null;
+            }
+
+            public override string ToStringValue()
+            {
+                return "forloop";
+            }
+
+            public override ValueTask<FluidValue> GetValueAsync(string name, TemplateContext context)
+            {
+                switch (name)
+                {
+                    case "length": return new ValueTask<FluidValue>(new NumberValue(Length));
+                    case "index": return new ValueTask<FluidValue>(new NumberValue(Index));
+                    case "index0": return new ValueTask<FluidValue>(new NumberValue(Index0));
+                    case "rindex": return new ValueTask<FluidValue>(new NumberValue(RIndex));
+                    case "rindex0": return new ValueTask<FluidValue>(new NumberValue(RIndex0));
+                    case "first": return new ValueTask<FluidValue>(new BooleanValue(First));
```

Comment:
`BooleanValue.Create(bool)` resolving to `BooleanValue.True` or `BooleanValue.False`?